File: n2/2.swift
import Foundation
print("Введите кол-во полных банок и эквивалент обмена: ", terminator: "")
if let line = readLine() {
   var cups = line.split(separator: " ").map { Int($0)! } // [полные, обмен]
    var drunk = cups[0] // изначально выпито
    var i = 0           // счётчик шагов
    var full: Int       // сколько полных банок в обмене

    // пока можно обменивать
    while cups[0] >= cups[1] {
        i += 1
        full = cups[0] / cups[1] // обмен
        i += 1
        let empty = cups[0] % cups[1] // остаток пустых
        drunk += full
        cups[0] = full + empty // обновляем банки
    }

    i += 1 // допиваем
    print("Выпито банок и кол-во итераций: \(drunk) \(i)")
}

File: n2/2.ts
import * as readline from 'readline';

const rl = readline.createInterface({
	input: process.stdin,
	output: process.stdout
});

rl.question("Введите кол-во полных банок и эквивалент обмена: ", (input) => {
	if (!input.trim()) {
		console.error("Нет ввода");
		rl.close();
		return;
	}	
	const cups = input.trim().split(/\s+/).map(Number);
    
	let drunk = cups[0];
	let i = 0;

	while (cups[0] >= cups[1]) {
		i++; // шаг опустошения
		const full = Math.floor(cups[0] / cups[1]); // полные банки после обмена
		i++; // шаг обмена
		drunk += full; // увеличиваем счётчик выпитых
		cups[0] = full + (cups[0] % cups[1]); // новые полные + оставшиеся пустые
	}

	i++; // финальное опустошение
	console.log(`Выпито банок и кол-во итераций: ${drunk} ${i}`);
    
	rl.close();
});
